src/Coriolis/general_f_plane.jl

```diff
@@ -0,0 +1,63 @@
+using Oceananigans.Grids: ZDirection, validate_unit_vector
+
+"""
+    GeneralFPlane{FT} <: AbstractRotation
+
+A Coriolis implementation that accounts for the locally vertical and
+both local horizontal components of the rotation vector. Traditionally
```

Comment:
maybe add 'possibly` before both, as we might have 2, 1 or 0?

Reply:
Done!